utils/ffmpegvideodecode/ffmpeg_video_dec.cpp

```diff
@@ -0,0 +1,866 @@
+            AVPacket *pkt = av_packet_alloc();
+            pkt->data = static_cast<u_int8_t *> (av_packet_data_[i].first);
+            pkt->size = av_packet_data_[i].second;
+            av_packets_.push_back(pkt);
+        }
+    }
+
+    if (curr_video_format_ptr_ == nullptr) {
+        curr_video_format_ptr_ = new RocdecVideoFormat();
+    }
+    // store current video format: this is required to call reconfigure from application in case of random seek
+    if (curr_video_format_ptr_) memcpy(curr_video_format_ptr_, p_video_format, sizeof(RocdecVideoFormat));
+
+    if (coded_width_ && coded_height_) {
+        end_of_stream_ = false;
+        // rocdecCreateDecoder() has been called before, and now there's possible config change
+        return ReconfigureDecoder(p_video_format);
+    }
+    // e_codec has been set in the constructor (for parser). Here it's set again for potential correction
+    codec_id_ = p_video_format->codec;
+    video_chroma_format_ = p_video_format->chroma_format;
+    bitdepth_minus_8_ = p_video_format->bit_depth_luma_minus8;
+    byte_per_pixel_ = bitdepth_minus_8_ > 0 ? 2 : 1;
+
+    // convert AVPixelFormat to rocDecVideoChromaFormat
+    video_surface_format_ = AVPixelFormat2rocDecVideoSurfaceFormat(decoder_pixel_format_);
+    coded_width_ = p_video_format->coded_width;
+    coded_height_ = p_video_format->coded_height;
+    disp_rect_.top = p_video_format->display_area.top;
+    disp_rect_.bottom = p_video_format->display_area.bottom;
+    disp_rect_.left = p_video_format->display_area.left;
+    disp_rect_.right = p_video_format->display_area.right;
+    disp_width_ = p_video_format->display_area.right - p_video_format->display_area.left;
+    disp_height_ = p_video_format->display_area.bottom - p_video_format->display_area.top;
+
+    // AV1 has max width/height of sequence in sequence header
+    if (codec_id_ == rocDecVideoCodec_AV1 && p_video_format->seqhdr_data_length > 0) {
+        // dont overwrite if it is already set from cmdline or reconfig.txt
+        if (!(max_width_ > p_video_format->coded_width || max_height_ > p_video_format->coded_height)) {
+            RocdecVideoFormatEx *vidFormatEx = (RocdecVideoFormatEx *)p_video_format;
+            max_width_ = vidFormatEx->max_width;
+            max_height_ = vidFormatEx->max_height;
+        }
+    }
+    if (max_width_ < (int)p_video_format->coded_width)
+        max_width_ = p_video_format->coded_width;
+    if (max_height_ < (int)p_video_format->coded_height)
+        max_height_ = p_video_format->coded_height;
+
+    if (!(crop_rect_.right && crop_rect_.bottom)) {
+        target_width_ = (disp_width_ + 1) & ~1;
+        target_height_ = (disp_height_ + 1) & ~1;
+    } else {
+        target_width_ = (crop_rect_.right - crop_rect_.left + 1) & ~1;
+        target_height_ = (crop_rect_.bottom - crop_rect_.top + 1) & ~1;
+    }
+
+    chroma_height_ = (int)(ceil(target_height_ * GetChromaHeightFactor(video_surface_format_)));
```

Comment:
formatting: please use `static_cast<int>(  )` here and in lines 284 and 288 below.